tests/test_architecture.py

```diff
@@ -100,6 +101,7 @@ def test_modeling_languages_should_not_depend_on_ui_package():
         archrule("Modeling languages should not depend on the UI package")
         .match("gaphor.C4Model*", "gaphor.RAAML*", "gaphor.SysML*", "gaphor.UML*")
         .should_not_import("gaphor.ui*")
+        .may_import("gaphor.ui.styleeditor")
```

Comment:
This rule should not be needed anymore.

If we allow (indirect) import of UI components that means that importing the UML/SysML packages does require GTK and libadwaita. This is something we do not want, since it makes it harder to use Gaphor as a library (without UI). By looking at the code it looks like it's not imported directly, only in the property pages, which are loaded as `gaphor.modules`.

Reply:
Hi @amolenaar, I've deleted the rule, but since `gaphor.ui.styleeditor` will be (non directly) imported by property pages, I've added a flag to only check direct imports (`only_direct_imports=True`).

Reply:
Ah, that's because I suggested to move the editor to the UI package. My bad, sorry 🙈 .

I pushed a commit that consolidates the style editor code (both property page and editor window) in one file, imported as module (you may need to `poetry install` for the module loaded).

I noticed that the style editor does not work when you open an existing model: it's linking `self.diagram.styleSheet` in the constructor, but during load time, no relations have been created yet when the presentation is created.

Given the number of fields we currently have in the editor, it may look nicer to use a `Gtk.Popover` instead of a window. Then the popover window can be linked to the editor button (with a dropdown icon). Also the text needs to be nicely alighed.

I also noticed the items do not update if a color is changed. I think we should trigger an update for the items on the current diagram, so they redraw with the new style.

Reply:
> I noticed that the style editor does not work when you open an existing model: it's linking `self.diagram.styleSheet` in the constructor, but during load time, no relations have been created yet when the presentation is created.
> 
Yes it is linking in the constructor and then in the load method of Presentation we perform a "name_change" so it restores the name.